stix_shifter_utils/stix_transmission/utils/RestApiClientAsync.py

```diff
@@ -87,8 +88,13 @@ async def call_api(self, endpoint, method, headers=None, cookies=None, data=None
                     except IOError:
                         self.logger.error('Failed to setup certificate')
 
+            if self.ssl_context and self.server_cert_file_content:
                 self.ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS)
-                self.ssl_context.load_verify_locations(self.server_cert_name)
+                try:
+                    self.ssl_context.load_verify_locations(self.server_cert_name)
+                except Exception as ex:
+                    self.logger.debug('Unable to load the certificate for ssl context. Reasons: Connection does not require certificate or unexpected exception while loading the certificate: ' + str(ex))
+
                 self.ssl_context.verify_mode = ssl.CERT_REQUIRED
```

Comment:
verify_mode and check_hostname cannot be conditional they must be applied even in the case of using a default ssl context but I don't see a call to `ssl.create_default_context()` any where.

Reply:
updated the code to remove ambiguity around certificate validation. now certificate is verified in both custom or default cases. also remove the option to force validation to false.